code/modules/reagents/reagent_containers/food/drinks/juicebox.dm

```diff
@@ -16,32 +68,63 @@
 /obj/item/chems/food/drinks/juicebox/open(mob/user)
 	playsound(loc,'sound/effects/bonebreak1.ogg', rand(10,50), 1)
 	to_chat(user, SPAN_NOTICE("You pull off the straw and stab it into \the [src], perforating the foil!"))
-	icon_state = "[initial(icon_state)]_open"
 	atom_flags |= ATOM_FLAG_OPEN_CONTAINER
+	appearance = update_juicebox()
 
 /obj/item/chems/food/drinks/juicebox/apple
 	name = "apple juicebox"
 	desc = "A small cardboard juicebox with a cartoon apple on it."
-	icon_state = "juicebox_red"
 
 /obj/item/chems/food/drinks/juicebox/apple/Initialize()
 	. = ..()
+	appearance = SetColors("#ff0000", "#ffff00", "#ff0000", style="stripe")
 	reagents.add_reagent(/datum/reagent/drink/juice/apple, 25)
 
 /obj/item/chems/food/drinks/juicebox/orange
 	name = "orange juicebox"
 	desc = "A small cardboard juicebox with a cartoon orange on it."
-	icon_state = "juicebox_orange"
 
 /obj/item/chems/food/drinks/juicebox/orange/Initialize()
 	. = ..()
+	appearance = SetColors("#ffff00", "#ff0000", "#ffff00", style="stripe")
 	reagents.add_reagent(/datum/reagent/drink/juice/orange, 25)
 
 /obj/item/chems/food/drinks/juicebox/grape
 	name = "grape juicebox"
 	desc = "A small cardboard juicebox with some cartoon grapes on it."
-	icon_state = "juicebox_purple"
 
 /obj/item/chems/food/drinks/juicebox/grape/Initialize()
 	. = ..()
+	appearance = SetColors("#ff00ff", "#00ff00", style="stripe")
 	reagents.add_reagent(/datum/reagent/drink/juice/grape, 25)
+
+/obj/item/chems/food/drinks/juicebox/random/Initialize()
+	. = ..()
+	var/primary = rgb(rand(0, 255), rand(0, 255), rand(0, 255))
```

Comment:
Global proc `get_random_colour()` uses 0-255 as defaults and returns a hex string.

Reply:
Definitely better